adapter/internal/oasparser/model/api_yaml.go

```diff
@@ -97,17 +97,39 @@ type OperationYaml struct {
 
 // OperationPolicies holds policies of the APIM operations
 type OperationPolicies struct {
-	In    []Policy `json:"in,omitempty"`
-	Out   []Policy `json:"out,omitempty"`
-	Fault []Policy `json:"fault,omitempty"`
+	In    PolicyList `json:"in,omitempty"`
+	Out   PolicyList `json:"out,omitempty"`
+	Fault PolicyList `json:"fault,omitempty"`
+}
+
+// policyStats used to optimize and reduce loops by storing stats by calculating only once
+type policyStats struct {
+	firstIndex int
+	count      int
+}
+
+// PolicyList holds list of Polices in a flow of operation
+type PolicyList []Policy
+
+func (pl PolicyList) getStats() map[string]policyStats {
```

Comment:
Shall we move this file after the constructor method for the apiyaml, and after the necessary methods for a regular API deployment. Let's go priority wise.

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/2656/commits/321b5c37e580d65072fbcd8902e950a87217755a